util/helm/client.go

```diff
@@ -390,10 +391,10 @@ func getTagsListURL(rawURL string, chart string) (string, error) {
 	if err != nil {
 		return "", fmt.Errorf("unable to parse repo url: %v", err)
 	}
+	tagsPathFormat := "%s/v2/%s/tags/list"
 	repoURL.Scheme = "https"
-	tagsList := strings.Join([]string{"v2", url.PathEscape(chart), "tags/list"}, "/")
-	repoURL.Path = strings.Join([]string{repoURL.Path, tagsList}, "/")
-	repoURL.RawPath = strings.Join([]string{repoURL.RawPath, tagsList}, "/")
+	repoURL.Path = fmt.Sprintf(tagsPathFormat, repoURL.Path, chart)
+	repoURL.RawPath = fmt.Sprintf(tagsPathFormat, repoURL.RawPath, url.PathEscape(chart))
```

Comment:
As per the documentation on `url.String()` and `url.EscapedPath()`:

> EscapedPath returns the escaped form of u.Path. In general there are multiple possible escaped forms of any path. EscapedPath returns u.RawPath when it is a valid escaping of u.Path. Otherwise EscapedPath ignores u.RawPath and computes an escaped form on its own.

This means that `repoURL.RawPath` should contain a valid escaping.
In the case of a `chart` with a `/` in its name, the following occurred:

- `url.PathEscape("chart/name")` generated `chart%2Fname`
- `repoURL.Path` and `repoURL.RawPath` **both** contain `chart%2Fname`
- `repoURL.String()` invokes `repoURL.EscapedPath()`
- `repoURL.EscapedPath()` does not recognize `chart%2Fname` as a valid escaping of `chart%2Fname`
- `repoURL.EscapedPath()` escapes again, resulting in `chart%252Fname`

and the resulting URL is not correct and results in 404 responses